compiler-rt/test/asan/TestCases/Posix/ignore_free_hook.cpp

```diff
@@ -0,0 +1,119 @@
+// Check that free hook doesn't conflict with Realloc.
+// RUN: %clangxx_asan -O2 %s -o %t -DTEST=basic_hook_works && not %run %t \
+// RUN:   |& FileCheck %s -check-prefix=CHECK-BASIC
+// RUN: %clangxx_asan -O2 %s -o %t -DTEST=ignore && %run %t \
```

Comment:
can this be sanitizer common test with
REQUIRE: asan || hwasan

Reply:
Seems reasonable and do-able except it gets kinda messy:

 1. Allocation type mismatch doesn't exist for HWASan and it would mean therefore forking that test into two files anyway.
 2. There aren't sanitizer_common tests in gn, only check-hwasan/check-asan, and hwasan tests are most important and easiest to invoke for android through gn.